key_test.go

```diff
@@ -40,24 +40,24 @@ func TestEncodeKey(t *testing.T) {
 	}
 }
 
-// TestDecodeRelativeKey tests the decodeRelativeKey function.
-func TestDecodeRelativeKey(t *testing.T) {
+// TestDecodeKey tests the decodeKey function.
+func TestDecodeKey(t *testing.T) {
 	tests := []struct {
 		key             string
 		expectedNetwork string
 	}{
 		{
-			key:             "10.4.0.0-16",
+			key:             "/ipam/subnet/10.4.0.0-16",
 			expectedNetwork: "10.4.0.0/16",
 		},
 		{
-			key:             "192.168.1.0-24",
+			key:             "/ipam/subnet/192.168.1.0-24",
```

Comment:
Why is all this changing now? I can imagine that we need an upgrade path here because the existing clusters are created based on certain assumptions of the lib so far. Changing this now without being backwards compatible would screw all AWS clusters. This probably also applies to the already merged PR.

Reply:
Look at

```diff
-		existingSubnetString := decodeRelativeKey(kv.Val())
+		existingSubnetString := decodeKey(kv.Key())
```

Now the value is retrieved from key (which didn't change). The value is now free form annotation (set by user), so we can't rely on it anymore.